### PR TITLE
Support using +Symbol+ to updating the environment.

### PR DESCRIPTION
**Pull request by hanhanW**
* Support using +Symbol+ to updating the environment.
* Add the documentations.

---
This change is [Reviewable](https://reviewable.io/reviews/peter50216/pwntools-ruby/57)

### CODE REVIEW

File: lib/pwnlib/context.rb [hunk 1]
```diff
@@ -174,17 +241,35 @@ def log_level=(value)
...
-        os = os.downcase
+        os = os.to_s.downcase
         raise ArgumentError, "os must be one of #{OSES.sort.inspect}" unless OSES.include?(os)
         @attrs[:os] = os
       end
 
+      # Set the signedness for packing opreation.
+      #
+      # @param [String, Symbol, true, false] value
+      #   The signedness.
```
Comment (david942j) [inline lib/pwnlib/context.rb]: reference the constant `SIGNEDNESSES` here to let users know what are valid arguments

File: lib/pwnlib/context.rb [hunk 2]
```diff
@@ -125,42 +150,84 @@ def clear
...
+      #
+      # @param [Integer] bytes
+      #   The word size.
       def bytes=(bytes)
         self.bits = bytes * 8
       end
 
+      # The endianness of the target machine.
+      #
+      # @param [String, Symbol] endian
```
Comment (david942j) [inline lib/pwnlib/context.rb]: `:little, :big`

**Comment by hanhanW** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[lib/pwnlib/context.rb, line 118 at r1](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/57#-KwoQFIbAIDx5G1C3f9--r1-118:-KxPJM8lR-XnEarZCrxb:b-896fix) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/726a43d489440cbf886a7cab94d81959c1ed45e8/lib/pwnlib/context.rb#L118)):*
<details><summary><i>Previously, david942j (david942j) wrote…</i></summary><blockquote>

`block returned`
</blockquote></details>

Done.

---

*[lib/pwnlib/context.rb, line 213 at r1](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/57#-KwoQFIbAIDx5G1C3f9--r1-213:-KxPJO90m4z2clNPSWth:b-m1mstc) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/726a43d489440cbf886a7cab94d81959c1ed45e8/lib/pwnlib/context.rb#L213)):*
<details><summary><i>Previously, david942j (david942j) wrote…</i></summary><blockquote>

`:little, :big`
</blockquote></details>

not only these... use ref to `ENDIANNESSES` instead.

---

*[lib/pwnlib/context.rb, line 260 at r1](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/57#-KwoQFIbAIDx5G1C3f9--r1-260:-KxPJYaSMl3jWktx_TZZ:b-896fix) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/726a43d489440cbf886a7cab94d81959c1ed45e8/lib/pwnlib/context.rb#L260)):*
<details><summary><i>Previously, david942j (david942j) wrote…</i></summary><blockquote>

reference the constant `SIGNEDNESSES` here to let users know what are valid arguments
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/57)*

**Comment by david942j** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1.
Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/57#-:-KxWTEwAND-Jw8zYbBJQ:bnfp4nl)*

**Comment by david942j** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/57)*

**Pull request merged**